tests/keras/test_keras_model_export.py

```diff
@@ -114,36 +127,37 @@ def keras_custom_env(tmpdir):
     return conda_env
 
 
-@pytest.mark.large
-def test_model_save_load(model, model_path, data, predicted):
+@pytest.mark.parametrize("build_model", [model, tf_keras_model])
+def test_model_save_load(build_model, model_path, data):
     x, _ = data
-    mlflow.keras.save_model(model, model_path)
-
+    keras_model = build_model(data)
+    if build_model == tf_keras_model:
+        model_path = os.path.join(model_path, "tf")
+    else:
+        model_path = os.path.join(model_path, "plain")
+    expected = keras_model.predict(x)
+    mlflow.keras.save_model(keras_model, model_path)
     # Loading Keras model
     model_loaded = mlflow.keras.load_model(model_path)
-    assert all(model_loaded.predict(x) == predicted)
-
+    assert type(keras_model) == type(model_loaded)
+    assert all(expected == model_loaded.predict(x))
     # Loading pyfunc model
     pyfunc_loaded = mlflow.pyfunc.load_pyfunc(model_path)
-    assert all(pyfunc_loaded.predict(x).values == predicted)
+    assert all(pyfunc_loaded.predict(x).values == expected)
 
     # pyfunc serve
     scoring_response = pyfunc_serve_and_score_model(
         model_uri=os.path.abspath(model_path),
         data=pd.DataFrame(x),
         content_type=pyfunc_scoring_server.CONTENT_TYPE_JSON_SPLIT_ORIENTED)
-    assert np.allclose(
-            pd.read_json(scoring_response.content, orient="records").values.astype(np.float32),
-            predicted,
-            rtol=1e-5,
-            atol=1e-9)
-
+    assert all(pd.read_json(scoring_response.content, orient="records").values.astype(np.float32)
+               == expected)
```

Comment:
I'm concerned about using a test for exact equality, given that we previously had issues with flaky Keras tests due to numeric precision problems. Can we stick with the `allclose` call?

Reply:
I don't think there is any need to lower the precision if this is working now. 
It this stops failing in the future I think we should investigate it. I mean, it may be ok if the precision fluctuates, but going from 100% equal to almost equal is a big step down.